### PR TITLE
Support logging more Go types to ETW

### PR DESCRIPTION
**Pull request by kevpar**
Signed-off-by: Kevin Parsons <kevpar@ntdev.microsoft.com>

### CODE REVIEW

File: internal/etw/fieldopt.go [hunk 1]
```diff
@@ -28,6 +60,313 @@ func StringArray(name string, values []string) FieldOpt {
...
+	default:
+		panic("Unsupported int size")
+	}
+}
+
+// IntArray adds an array of int to the event.
+func IntArray(name string, values []int) FieldOpt {
+	inType := InTypeNull
+	var writeItem func(*EventData, int)
+	switch unsafe.Sizeof(values[0]) {
```
Comment (jterry75) [inline internal/etw/fieldopt.go]: This will panic if len 0

File: internal/etw/fieldopt.go [hunk 2]
```diff
@@ -28,6 +60,313 @@ func StringArray(name string, values []string) FieldOpt {
...
+	default:
+		panic("Unsupported uint size")
+	}
+}
+
+// UintArray adds an array of uint to the event.
+func UintArray(name string, values []uint) FieldOpt {
+	inType := InTypeNull
+	var writeItem func(*EventData, uint)
+	switch unsafe.Sizeof(values[0]) {
```
Comment (jterry75) [inline internal/etw/fieldopt.go]: Same will panic on empty

File: internal/etw/fieldopt.go [hunk 3]
```diff
@@ -28,6 +60,313 @@ func StringArray(name string, values []string) FieldOpt {
...
+		em.WriteField(name, inType, OutTypeDefault, 0)
+		writeItem(ed, value)
+	}
+}
+
+// UintptrArray adds an array of uintptr to the event.
+func UintptrArray(name string, values []uintptr) FieldOpt {
+	inType := InTypeNull
+	var writeItem func(*EventData, uintptr)
+	switch unsafe.Sizeof(values[0]) {
```
Comment (jterry75) [inline internal/etw/fieldopt.go]: And again.

File: pkg/etwlogrus/hook_test.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+			t.Fatal(err)
+		}
+	}()
+
+	fireEvent(t, p, "Bool", true)
+	fireEvent(t, p, "BoolSlice", []bool{true, false, true})
+	fireEvent(t, p, "String", "teststring")
+	fireEvent(t, p, "StringSlice", []string{"sstr1", "sstr2", "sstr3"})
+	fireEvent(t, p, "Int", int(1))
+	fireEvent(t, p, "IntSlice", []int{2, 3, 4})
```
Comment (jterry75) [inline pkg/etwlogrus/hook_test.go]: Add an empty test for all the slices

File: internal/etw/fieldopt.go (hunk 1 above)
Comment (kevpar) [reply internal/etw/fieldopt.go]: This actually seems to work, due to unsafe.Sizeof being a compile-time constant expression. The tests I'm adding to log empty slices of int, uint, and uintptr all pass.

**Comment by kevpar** [comment]
Going to handle array support in a future PR.

**Pull request merged**